src/components/cc-stretch/cc-stretch.js

```diff
@@ -0,0 +1,105 @@
+import { css, html, LitElement } from 'lit';
+
+/**
+ * A layout component used to switch visibility between several elements while preventing layout shifts.
+ *
+ * Use `visibleElementId` prop to provide the `id` value corresponding to the element you want to make visible.
+ * Other elements are hidden with `visibility: hidden`.
+ *
+ * More information about the issues this component solves and the technique it uses can be found in [Hubert Sablonnière's blog Post - "Prevent layout shifts with CSS grid stacks"](https://www.hsablonniere.com/prevent-layout-shifts-with-css-grid-stacks--qcj5jo/).
+ *
+ * **Caution**:
+ *
+ * This component relies on a class named `cc-stretch--visible` applied to the relevant slotted element. Make sure you never override this class.
+ *
+ * @cssdisplay inline-grid
+ *
+ * @slot - The containers you want to switch between. The container you want to be displayed must have an `id` attribute matching the value of the `visibleElementId` prop.
+ *
+ * @cssprop {AlignItems} --cc-stretch-align-items - Specify how the items should be distributed / positioned vertically within the grid (Default: `center`. Possible values: https://developer.mozilla.org/en-US/docs/Web/CSS/align-items).
+ * @cssprop {JustifyItems} --cc-stretch-justify-items - Specify how the items should be distributed / positioned horizontally within the grid (Default: `center`. Possible values: https://developer.mozilla.org/en-US/docs/Web/CSS/justify-items).
+ */
+
+export class CcStretch extends LitElement {
+
+  static get properties () {
+    return {
+      disableStretching: { type: Boolean, attribute: 'disable-stretching', reflect: true },
+      visibleElementId: { type: String, attribute: 'visible-element-id', reflect: true },
```

Comment:
Seems like we don't need this one to have `reflect`, WDYT?

Reply:
Well, I didn't at first, then @pdesoyres-cc suggested reflecting it and I figured why not:
if developers want to style the box with a different background when a specific element is displayed ? (writing this makes me think it's far fetched actually but it does not hurt ? :joy:).
WDYT ?